### ISSUE TITLE
[Issue]: slow generation compared to auto

### ISSUE THREAD

**Issue by gsgoldma** [vladmandic/automatic#261]
### Issue Description

I'm averaging about > 2.x it/s
in auto I get about 1.56 it/s

settings seem to identical 

### Version Platform Description

windows 10

**Comment by vladmandic** [comment]
Use system info and run benchmark and then submit reports for both here. Otherwise performance numbers that are thar close are not reliable.
And you haven't provided and information on your platform.

**Comment by gsgoldma** [comment]
I made a typo, it was supposed to be 2.x s/i -my bad

using xformers and gpu is 980 ti 6gigs, and I'll find out torch later today/run benchmarks

**Comment by vladmandic** [comment]
> using xformers and gpu is  nvidia 980 ti 6gigs, and I'll find out torch later today/run benchmarks

Are you sure you're using xformers?
In my repo, xformers are not enabled by default (SDP is default) even if you have them installed and use cmd flags - you need to select them in settings.

All that information is necessary when you're creating an issue like this. Having issue that says "it's slower" without providing any info cannot result in any progress.

**Comment by gsgoldma** [comment]
venv "D:\automatic\venv\Scripts\Python.exe"
13:02:08-108533 INFO     Python 3.10.9 on Windows
13:02:08-454423 INFO     Running setup
13:02:08-496425 INFO     Version: e6701f5c Wed Apr 19 14:12:19 2023 -0400
13:02:09-058452 INFO     Updating Wiki
13:02:10-892791 INFO     Installing requirements
13:02:13-094699 INFO     Installing packages
13:02:13-097699 INFO     Installing repositories
13:02:13-302699 INFO     Installing submodules
13:02:17-904786 INFO     Updating submodules
13:02:32-530210 INFO     Extensions enabled: ['a1111-sd-webui-lycoris', 'clip-interrogator-ext', 'LDSR', 'Lora',
                         'multidiffusion-upscaler-for-automatic1111', 'prompt-bracket-checker', 'ScuNET',
                         'sd-dynamic-thresholding', 'sd-extension-aesthetic-scorer', 'sd-extension-steps-animation',
                         'sd-extension-system-info', 'sd-webui-controlnet', 'sd-webui-model-converter', 'seed_travel',
                         'stable-diffusion-webui-images-browser', 'stable-diffusion-webui-rembg', 'SwinIR']
13:03:35-537016 INFO     Extensions enabled: ['canvas-zoom', 'depth-image-io-for-SDWebui', 'depthmap2mask', 'gif2gif',
                         'infinite-zoom-automatic1111-webui', 'model-keyword', 'openOutpaint-webUI-extension',
                         'openpose-editor', 'PBRemTools', 'posex', 'prompt-fusion-extension', 'sd-3dmodel-loader',
                         'sd-canvas-editor', 'SD-latent-mirroring', 'sd-webui-3d-open-pose-editor',
                         'sd-webui-additional-networks', 'sd-webui-ar', 'sd-webui-aspect-ratio-helper',
                         'sd-webui-infinite-image-browsing', 'sd-webui-llul', 'sd-webui-panorama-viewer',
                         'sd-webui-regional-prompter', 'sd-webui-text2video', 'sd_save_intermediate_images',
                         'shift-attention', 'stable-diffusion-NPW', 'stable-diffusion-webui-aesthetic-gradients',
                         'stable-diffusion-webui-anti-burn', 'Stable-Diffusion-Webui-Civitai-Helper',
                         'stable-diffusion-webui-daam', 'stable-diffusion-webui-depthmap-script',
                         'stable-diffusion-webui-pixelization', 'stable-diffusion-webui-two-shot',
                         'ultimate-upscale-for-automatic1111', 'video_loopback_for_webui']
13:04:29-681816 INFO     Setting environment tuning
13:04:31-762821 INFO     Torch 2.0.0+cu118
13:04:31-789816 INFO     Torch backend: nVidia CUDA 11.8 cuDNN 8700
13:04:31-792820 INFO     Torch detected GPU: NVIDIA GeForce GTX 980 Ti VRAM 6144 Arch (5, 2) Cores 22
13:04:31-798820 INFO     Server arguments: []
Available models: D:\stable-diffusion-webui\models\Stable-diffusion 456
[AddNet] Updating model hashes...
0it [00:00, ?it/s]
[AddNet] Updating model hashes...
0it [00:00, ?it/s]
Civitai Helper: Get Custom Model Folder
Civitai Helper: Load setting from: D:\automatic\extensions\Stable-Diffusion-Webui-Civitai-Helper\setting.json
Loading theme: gradio/seafoam
1280 720
670
Running on local URL:  http://127.0.0.1:7860

**Comment by vladmandic** [comment]
How about you collect requested info and post that once available? Updating issue with startup log is also not helping, other than you have 10+ extensions that may or may not have performance impact.

**Comment by gsgoldma** [comment]
"2023-04-19 13:29:21.715362",
    "0.54 / 0.55 / 0.55",
    "updated:2023-04-19 hash:e6701f5c url:https://github.com/vladmandic/automatic.git/tree/master",
    "arch:AMD64 cpu:Intel64 Family 6 Model 60 Stepping 3, GenuineIntel system:Windows release:Windows-10-10.0.19044-SP0 python:3.10.9",
    "torch:2.0.0+cu118 autocast  half xformers:0.0.17 accelerate:0.18.0 transformers:4.26.1",
    "device:NVIDIA GeForce GTX 980 Ti (1) (compute_37) (5, 2) cuda:11.8 cudnn:8700 6GB",
    "xformers none",
    "SD 1.5 Models\\v1-5-pruned.ckpt [e1441589a6]",
    "SD",
    "",
    "1dafb8"

I think this is what you're asking for. I ran the benchmark in system info then went to the json file it says it was saved to.

**Comment by vladmandic** [comment]
From just initial view, you don't have xformers enabled.
Normally, my default choice for cross-optimization is best-suited for most GPUs - but that is unless you have a very old GPU in which case it is fully expected that performance would be better with xformers. And I wrote about that in numerous posts.

So enable xformers and retry?

> I think this is what you're asking for

Yes, but i need it for both so i can compare

> The startup is indeed slowed a bit by the extensions

Its not just the startup, some extensions impact either prompt parsing (model-keyword, sd-dynamic-prompt) so there is overhead before generation starts so it appears slower. And some extensions impact performance on each step. I'm not blaming any specific extension, but if we're going to actually get to the bottom of this, its always best to minimize number of variables. And extensions add a lot to complexity.

So, to recap: disable all user extensions and run benchmark for both A1111 and my repo and post results here.

**Comment by gsgoldma** [comment]
[
  [
    "2023-04-19 13:29:21.715362",
    "0.54 / 0.55 / 0.55",
    "updated:2023-04-19 hash:e6701f5c url:https://github.com/vladmandic/automatic.git/tree/master",
    "arch:AMD64 cpu:Intel64 Family 6 Model 60 Stepping 3, GenuineIntel system:Windows release:Windows-10-10.0.19044-SP0 python:3.10.9",
    "torch:2.0.0+cu118 autocast  half xformers:0.0.17 accelerate:0.18.0 transformers:4.26.1",
    "device:NVIDIA GeForce GTX 980 Ti (1) (compute_37) (5, 2) cuda:11.8 cudnn:8700 6GB",
    "xformers none",
    "SD 1.5 Models\\v1-5-pruned.ckpt [e1441589a6]",
    "SD",
    "",
    "1dafb8"
  ]
Results from Auto, apologies about the delay

**Comment by vladmandic** [comment]
with xformers:
- my repo: "0.54 / 0.55 / 0.55" using torch 2.0
- a1111: "0.54 / 0.55 / 0.55" using torch 1.13
i do not see any differences in performance.
with doggettx:
- a1111: "1.21 / 1.55 / 1.58"
basicaly 2.5x faster. so set doggettx as your chosen cross-optimization here - that is the glaring difference.
btw, numbers are iterations per second for batch sizes 1, 2 and 4.
